### DISCUSSION TITLE
Long line wrap problem for chinese characters

### DISCUSSION THREAD

**Discussion by amitbha** [orbitalquark/textadept#332] [Q&A] [answered]
`view.wrap_mode = view.WRAP_CHAR` is set.
But TA seems not able to recognize the chinese character boundaries.
![ksnip_20230110-170503](https://user-images.githubusercontent.com/4060625/211508308-4187029a-08ce-47bb-99a0-072ee642ca50.png)
I highlight the problem line, which is wraped after a blank space, leaving very long blanks.
If the punctuations can be used as boundaries, the problem will be partially solved.

Text for test:
`> 以是之故，须生智慧，彼因亦于清净之经典，与心之能力相称而为听闻。《摄度论》云：“寡闻盲汉不知修，无闻彼当何所思，故当勤闻从彼因，如思以修广生慧。”至尊慈氏亦云：“三轮诸分别，许彼所知障。悭等诸分别，许彼烦恼障。远离彼诸障，除慧更无余。胜慧闻为本，是故闻为胜。”《集学论》亦云：“应忍当求闻，次应住林薮，当精进等引。”`

**Comment by orbitalquark** [comment, marked as answer]
Are you sure about your `view.wrap_mode` setting? I copy-pasted your 
test text and used the command entry to set `wrap_mode = WRAP_CHAR` and 
I appear to get proper wrapping. Any other wrap mode (WRAP_WORD or 
WRAP_WHITESPACE) gives me the same wrapping as in your screenshot (">" 
is on its own line). Note that Textadept's "Buffer > Toggle Wrap Mode" 
uses WRAP_WHITESPACE by default, so you'll get the same behavior as in 
your screenshot.

Wrapping has handled by Scintilla, Textadept's editing component, so any 
bugs would need to be reported upstream (either by you or me).

**Reply by amitbha** [reply]
Thks for the reply.
The problem is gone. I don't know why. Maybe the `view.wrap_mode` setting didn't function for some reason.
